### PR TITLE
Mecha Gamemode: Jockeys

### PR DESCRIPTION
**Pull request by Geevies**
* Added a new gamemode, Jockeys, featuring a new antag, Jockey. A mecha-antag focused gamemode. Minimum two antags, minimum ten players on the server for it to roll.

They start in a little antag base and teleport onto the ship similarly to how ninjas do. They have a workshop attached to their base with almost every piece of mecha equipment in the game available to them, so each player can construct their own mech they want to play with.

The jockeys themselves are equipped with a silenced pistol in a holster and a spare magazine in their pocket, and have a 10TC uplink in the other pocket.

To deal with sustained power issues, I've added min nuclear core power cells that store lots of power and regenerate 10% of it every 30 seconds.

image

I'm aware that there's a map freeze at the moment, I'll remap it later afterwards.

### CODE REVIEW

File: code/game/antagonist/outsider/jockey.dm [hunk 1]
```diff
@@ -0,0 +1,57 @@
+var/datum/antagonist/jockey/jockeys
+
+/datum/antagonist/jockey
+	id = MODE_JOCKEY
+	role_text = "Jockey"
+	role_text_plural = "Jockeys"
+	bantype = "jockey"
+	antag_indicator = "jockey"
+	landmark_id = "jockeyspawn"
+	welcome_text = "You are a Jockey, one of the best damn mech pilots in the spur. You've been sent here to cause mayhem.<br>\
```
Comment (Arrow768) [inline code/game/antagonist/outsider/jockey.dm:10]:
```suggestion
	welcome_text = "You are a Jockey, one of the best damn mech pilots in the spur.<br>\
```

**Comment by Geevies** [comment]
!review

File: code/game/antagonist/outsider/jockey.dm [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+/datum/antagonist/jockey
+	id = MODE_JOCKEY
+	role_text = "Jockey"
+	role_text_plural = "Jockeys"
+	bantype = "jockey"
+	antag_indicator = "jockey"
+	landmark_id = "jockeyspawn"
+	welcome_text = "You are a Jockey, one of the best damn mech pilots in the spur.<br>\
+	Your (syndicate) sponsored uplink will grant you access to various tools you may need to attempt to accomplish your goal.<br>\
```
Comment (SleepyGemmy) [inline code/game/antagonist/outsider/jockey.dm:11]:
```suggestion
	Your uplink will grant you access to various tools you may need to attempt to accomplish your goal.<br>\
```
people should be able to come up with their own motivations.

File: code/game/antagonist/outsider/jockey.dm [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	id_type = /obj/item/card/id/syndicate
+
+/datum/antagonist/jockey/New()
+	..()
+	jockeys = src
+
+/datum/antagonist/jockey/update_access(var/mob/living/player)
+	for(var/obj/item/storage/wallet/W in player.contents)
+		for(var/obj/item/card/id/id in W.contents)
+			id.name = "[player.real_name]'s Passport"
```
Comment (SleepyGemmy) [inline code/game/antagonist/outsider/jockey.dm:33]:
```suggestion
			id.name = "passport - [player.real_name]"
```

File: code/modules/heavy_vehicle/mecha.dm [hunk 4]
```diff
@@ -138,7 +138,7 @@
 	if(!user || !user.client)
 		return
 	to_chat(user, "That's \a <b>[src]</b>.")
-	to_chat(user, desc)
+	if(desc) to_chat(user, desc)
```
Comment (SleepyGemmy) [inline code/modules/heavy_vehicle/mecha.dm:141]:
```suggestion
	if(desc)
		to_chat(user, desc)
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by DreamySkrell — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**